rfc/0054-sdk3-management-apis.md

```diff
@@ -2523,6 +2863,142 @@ interface AnalyticsIndex{
+* One, and only one, of `dataverse` or `scope` is set.
+* `name` is set.
+* `hostname` is set.
+* When encryption level is set to "none" or "half":
+  * `username` is set.
+  * `password` is set.
+* When encryption level is set to "full":
+  * `certificate` is set.
+  * Either both `username` and `password` are set OR both `clientCertificate` and `clientKey` are set.
+
+The `LinkType()` implementation must return a value that corresponds to `"couchbase"`.
+
+### CouchbaseAnalyticsEncryptionSettings
+
+`CouchbaseAnalyticsEncryptionSettings` are the settings available for setting encryption level on an analytics link.
+
+* `encryption` (AnalyticsEncryptionLevel) - The encryption level to apply.
+* `certificate` (List<Byte>) - The certificate to use for the encryption when encryption level is set to "full".
+* `clientCertificate` (List<Byte>) - The client certificate to use for authenticating when encryption level is set to "full".
+* `clientKey` (List<Byte>) - The client key to use for authenticating when encryption level is set to "full".
+
+### S3ExternalAnalyticsLink
+
+`S3ExternalAnalyticsLink` provides a means of mapping external s3 analytics link details into an object.
+The following fields are named according to how they are sent when form/json encoded.
+The object names should be implemented in a way idiomatic to the implementing language.
+
+* `dataverse` (string) - The dataverse that the link belongs to.
+* `scope` (string) - The scope that the link belongs to. Only available in server 7.0+. Form is `bucket/scope`.
+* `name` (string) - The name of the link.
+* `accessKeyID` (string) - The AWS S3 access key.
+* `secretAccessKey` (string) - The AWS S3 secret key.
+* `sessionToken` (string) - The AWS S3 token if temporary credentials are provided. Only available in server 7.0+.
+* `region` (string) - The AWS S3 region.
+* `serviceEndpoint` - The AWS S3 service endpoint.
+
+A value for `type` of `s3` must be sent in the form payload, this is returned in the same manner in the JSON payload.
+This field is not exposed on the object and is either inferred from the object type or by calling `LinkType()`.
+
+The `Validate()` implementation must verify that:
+* One, and only one, of `dataverse` or `scope` is set.
+* `name` is set.
+* `accessKeyID` is set.
+* `secretAccessKey` is set.
+* `region` is set.
+
+The `LinkType()` implementation must return a value that corresponds to `"s3"`.
+
+### AzureBlobExternalAnalyticsLink
+
+`AzureBlobExternalAnalyticsLink` provides a means of mapping external azure analytics link details into an object.
+The `AzureBlobExternalAnalyticsLink` is available in server 7.0 Developer Preview mode and must be marked as `volatile` API stability.
+The following fields are named according to how they are sent when form/json encoded.
+The object names should be implemented in a way idiomatic to the implementing language.
+
+* `scope` (string) - The scope that the link belongs to. Form is `bucket/scope`.
```

Comment:
Does `AzureBlobExternalAnalyticsLink` have a `dataverse` property? If not, do we understand why?

Reply:
See the paragraph just above the fields for this link type. Azure support is 7.0 DP and dataverse is deprecated in 7.0, therefore I don't think that we should support dataverse for this link type. Note that this link type is also marked volatile.